src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -314,20 +314,16 @@ void NotationViewInputController::mousePressEvent(QMouseEvent* event)
         return;
     }
 
-    if (m_interactData.hitElement) {
-        if (!m_interactData.hitElement->selected()) {
-            if (event->button() != Qt::MouseButton::RightButton) {
-                SelectType selectType = SelectType::SINGLE;
-                if (keyState == Qt::NoModifier) {
-                    selectType = SelectType::SINGLE;
-                } else if (keyState & Qt::ShiftModifier) {
-                    selectType = SelectType::RANGE;
-                } else if (keyState & Qt::ControlModifier) {
-                    selectType = SelectType::ADD;
-                }
-                m_view->notationInteraction()->select({ m_interactData.hitElement }, selectType, m_interactData.hitStaffIndex);
-            }
+    if (m_interactData.hitElement && !m_interactData.hitElement->selected()) {
+        SelectType selectType = SelectType::SINGLE;
+        if (keyState == Qt::NoModifier) {
+            selectType = SelectType::SINGLE;
+        } else if (keyState & Qt::ShiftModifier) {
```

Comment:
Does it make sense for shift+right click to do a range selection and ctrl+right click to add to a list selection? @Tantacrul 
That's not what happens in, say, File Explorer in windows - right click always cancel the current selection and just selects the single file you click on.

Reply:
I need to test this to get an idea of how this feels. Can't quite comment on what I think is best - despite @wizofaus's comments, which are compelling. I'm also not on top of the collection of related issues. Perhaps @bkunda can advise / test.

Will resume normal services soon. A bit snowed under at the mo.

Reply:
From what I have been able to test here, right-click now correctly selects a new item and automatically triggers the context menu ✅ and left-clicking outside the context menu closes the menu ✅. So far so good 👍🏻

I'm not sure I completely understand all of @wizofaus's comments, particularly around shift+right click v. ctrl+right click. But as far as this is concerned:

"the implementation done here means that if you carefully select a list of items, then right click on one of them to access the context menu, you lose your selection. I don't think that would be expected."

...I was successful in selecting a range of non-adjacent items using ctrl+left click, then right-clicking a selected item (or indeed, right clicking a _new_ item with the ctrl key still pressed) to trigger the context menu, which then affected all of my selected items. This to me is expected behaviour, and it is consistent with how the File Explorer works (actually it's better than how File Explorer works, because File Explorer won't let you add a new item to the list and trigger the context menu at the same time). 

In saying this, I was only able to achieve this operation on single items. Something we seem not to be able to do (in neither 3.6 nor 4), which would be really useful, is to be able to perform the same add-to-list/non-adjacent selection on non-consecutive _ranges_ (i.e. multiple _bars_ of music at a time, not located next to each other). Perhaps this lies outside the scope of this particular issue, but since we're now discussing multiple selections, I wonder if it can be done. @RomanPudashkin?

Reply:
"Non-consecutive ranges" wouldn't really make sense to me, and I can't see how we could support them sensibly - definitely outside the scope of this PR anyway!
Did you test holding down CTRL (or SHIFT) to select multiple items, then releasing CTRL/SHIFT then right-clicking on one of the selected items?  From what I can see in the code, that will *deselect* all the existing items, which is not how other apps behave.

Reply:
This is what I have managed to doÖ
https://user-images.githubusercontent.com/86290556/126283619-5657f038-e188-48a8-a4d0-e9fc5b9e9096.mov

Of course the clef is not affected (this is consistent with MS3.6 at any rate).

Reply:
Ah yep...I just re-read the code, you're right, if you right-click on an already selected note it won't change the selection, my bad.

Reply:
@wizofaus you were right, there is a problem :) The selected area (it is important to use SHIFT for selecting) will be deselected if a user right-clicks on some elements from this area. I prepared a fix in the last commit. @bkunda could you check how the context menu and the elements selecting feel now (especially in cases when CTRL and SHIFT are used for that)?

Reply:
Almost there @RomanPudashkin!. Right-clicking on a selected area (whether it was selected by shift or ctrl) does not deselect the selection anymore. This is now working as expected. 

What isn't now working is: crash when shift+selecting or ctrl+selecting a range, then _right clicking_ outside of that range.   

Interestingly, the crash did **not** occur if the right click was in a different frame (e.g. a vertical frame containing header text).

This crash was repeatable in exactly the same way in both MacOS and Windows builds:

https://user-images.githubusercontent.com/86290556/126325124-abbd4c0d-1fba-4235-8ccd-8cc7626a21da.mov


https://user-images.githubusercontent.com/86290556/126325139-e04b0a6b-a9d3-48e9-94a0-01d337c6f568.mov


https://user-images.githubusercontent.com/86290556/126325149-f254d48f-c0a5-4592-ba83-5a98310f574a.mov

Reply:
@bkunda thanks for testing! Fixed